app/models/manageiq/providers/openstack/cloud_manager/provision_workflow.rb

```diff
@@ -1,8 +1,10 @@
 class ManageIQ::Providers::Openstack::CloudManager::ProvisionWorkflow < ::MiqProvisionCloudWorkflow
   def allowed_instance_types(_options = {})
-    source = load_ar_obj(get_source_vm)
-    ems = get_targets_for_ems(source, :cloud_filter, Flavor, 'flavors')
-    ems.each_with_object({}) { |f, h| h[f.id] = display_name_for_name_description(f) }
+    source                = load_ar_obj(get_source_vm)
+    flavors               = get_targets_for_ems(source, :cloud_filter, Flavor, 'flavors')
+    minimum_disk_required = source.hardware.size_on_disk
```

Comment:
@Ladas Is this the best value to use to determine the valid flavors for an image provision?  I am seeing images that have a size of 0 which means all flavors would be returned.  This is using the data you added in PR #5511.